transcom/mymove
# B 18823 ghc create requested office user int (DO NOT MERGE BEFORE INT B-18824)

## [B-18823](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A907135)

## Upstream

#12121 

## Summary

Introduces new GHC V1 endpoint `POST requested-office-users` to allow for office users to create new office users.
Introduces new service object for roles to be associated on user creation.
Updated Daniel’s create office user endpoint to also return roles

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Import latest ghc.yaml into postman
2. Grab a transportation office id from the transportation-offices GHC endpoint. Grab the “Id”, not the “Address -> Id”. There are 2 Ids per response back, grab the root one.
3. Login to office as any office user
4. Open network tab in browser dev tools and refresh page
5. Select latest network request, select headers
6. Grab ‘X-Csrf-Token’ value from the headers
7. Open postman
8. Navigate to MilMove GHC API requested-office-users “POST Create an Office User”
9. Navigate to Headers and add a new header, ‘X-Csrf-Token’, past your value in the value box
10. Update the URL to be respective of your environment, such as `officelocal:3000/ghc/v1/requested-office-users`
11. Update the body to be something like this (Update the transportationOfficeId:
```
{
  "firstName": "John",
  "lastName": "Doe",
  "email": "johndoe12@example.com",
  "telephone": "555-555-5555",
  "transportationOfficeId": "5a3388e1-6d46-4639-ac8f-a8937dc26938",
  "roles": [
    {
      "name": "Transportation Ordering Officer",
      "roleType": "transportation_ordering_officer"
    }
  ],
  "edipi": “1234456789",
  "otherUniqueId": "2234567892"
}
```
12. You should receive a return such as this
```
	{
    "active": false,
    "createdAt": "2024-03-06T17:12:41.777Z",
    "edipi": "1234456789",
    "email": "johndoe12@example.com",
    "firstName": "John",
    "id": "a2f04463-a7f7-4f8d-969f-590540c2ba13",
    "lastName": "Doe",
    "middleInitials": null,
    "otherUniqueId": "2234567893",
    "rejectionReason": null,
    "roles": [
        {
            "createdAt": "2024-03-04T13:22:01.338Z",
            "id": "ed095ebb-7291-4276-97df-3672df26b4ab",
            "roleName": "Transportation Ordering Officer",
            "roleType": "transportation_ordering_officer",
            "updatedAt": "2024-03-04T13:22:01.338Z"
        }
    ],
    "status": "REQUESTED",
    "telephone": "555-555-5555",
    "transportationOfficeId": "5a3388e1-6d46-4639-ac8f-a8937dc26938",
    "updatedAt": "2024-03-06T17:12:41.777Z",
    "userId": "1eea8c6c-3d35-454d-90e8-444fe95a205d"
}
```
13. Modify JSON body accordingly for test cases of validation
